app/Http/Controllers/EntryController.php

```diff
@@ -43,10 +44,33 @@ public function store(EntryStoreRequest $request)
     {
         $entryData = collect($request->validated())->except('items');
         $itemData = collect($request->validated())->only('items');
+
+        if ($itemData->has('items')) {
+            $itemData = $itemData['items'];
+        }
+
+        if ($itemData instanceof Collection) {
+            $itemData = $itemData->toArray();
+        }
+
+        //possibly implement as function for reusability
+        if ($entryData->get('choice') == 'advancement') {
+            // advancement: increment character's level
+            $entryData['levels'] = 1;
+            $itemData = [];
+        } elseif ($entryData->get('choice') == 'magic_item') {
+            // magic_item: attach item to character of choice, and set entry's levels to 0
+            $itemData = [$itemData[0] ?? []];
+            $entryData['levels'] = 0;
+        } elseif ($entryData->get('choice') == 'campaign_reward') {
+            // campaign reward: set levels = 0, no item(s), should contain custom note
+            $itemData = [];
+            $entryData['levels'] = 0;
+        }
+
```

Comment:
The more i stare at this the more i definitely think we're gonna need to refactor this into a seperate function of some sort, but it'll do for now

Reply:
Agreed, can be used for both update and store methods